Add `USERNAME` and `EMAIL` env variables

ozaner opened the pull request:
Added support for a custom username and email to be used in the autogenerated commit. Just set `USERNAME` and `EMAIL` respectively in the env variables.

## Context
The autogenerated commit this action produces always uses the username and email of the pusher. This commit adds support for changing these variables using the `USERNAME` and `EMAIL` environment variables. My particular use case was to mark these autogenerated commits as being created by the github-actions bot:

```yaml
- name: Commit to Output
  uses: s0/git-publish-subdir-action@develop
  env:
    REPO: self
    BRANCH: output
    FOLDER: outputDir
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    USERNAME: "github-actions[bot]"
    EMAIL: "github-actions[bot]@users.noreply.github.com"
```

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

s0 commented:
Hiya, thanks for the contribution, this is a nice addition.

Before we can get this merged in though, we need to add a few things to this PR:

* [ ] Update the README with instructions on how to use these parameters.
* [ ] Regenerate the action code (run `npm run build` from `action/`).
* [ ] Add unit-tests that cover the use of these parameters.

I can help you with any of the above or do it myself, but happy to let you have a go if you'd prefer to do it?

ozaner commented:
I've updated the readme, regenerated the action code, and added a unit test. Not too familiar with typescript (or even javascript) so I might be missing something.

promise commented:
@s0 bump

s0 commented on action/src/index.ts:
```diff
@@ -127,6 +127,16 @@ export interface EnvironmentVariables {
+  /**
+   * An optional string to use as the email on the git commit.
+   */
+  EMAIL?: string;
```
Can we change `EMAIL` to `COMMIT_EMAIL`

s0 commented on action/src/index.ts:
```diff
@@ -127,6 +127,16 @@ export interface EnvironmentVariables {
+  /**
+   * An optional string to use as the username on the git commit.
+   */
+  USERNAME?: string;
```
Can we change `USERNAME` to `COMMIT_NAME`

s0 reviewed (changes requested):
Hi! Thanks so much for this contribution, and for the docs and tests too, and sorry for the delayed review. I have some comments regarding naming, but other than this I think it's good to go! Thanks!

s0 commented:
(also sorry for not I clouding these change requests in the first review, I had some time to think about the naming and only remembered after I saw it in my notifications again)

ozaner commented:
The `USERNAME` and `EMAIL` env variables have been renamed, and the readme updated.
